editor/gui/scene_tree_editor.cpp

```diff
@@ -1548,16 +1572,34 @@ void SceneTreeDialog::set_valid_types(const Vector<StringName> &p_valid) {
 		HBoxContainer *hb = memnew(HBoxContainer);
 		hflow->add_child(hb);
 
+		// Attempt to get the correct name and icon for script path types
+		String type_name = type;
+		Ref<Texture2D> type_icon = EditorNode::get_singleton()->get_class_icon(type);
+
+		// If we can't find a global class icon, try to find one for the script.
+		if (type_icon.is_null()) {
+			if (ResourceLoader::exists(type, "Script")) {
+				Ref<Script> node_script = ResourceLoader::load(type);
+				if (node_script.is_valid()) {
+					type_name = type_name.get_file();
+					type_icon = EditorNode::get_editor_data().get_script_icon(node_script);
+					if (type_icon.is_null()) {
+						type_icon = EditorNode::get_singleton()->get_class_icon(node_script->get_class_name());
+					}
+				}
+			}
+		}
```

Comment:
```suggestion
		if (type_icon.is_null() && ResourceLoader::exists(type, "Script")) {
			Ref<Script> node_script = ResourceLoader::load(type);
			if (node_script.is_valid()) {
				type_name = type_name.get_file();
				type_icon = EditorNode::get_editor_data().get_script_icon(node_script);
				if (type_icon.is_null()) {
					type_icon = EditorNode::get_singleton()->get_class_icon(node_script->get_class_name());
				}
			}
		}
```
Compress `if` statements IMO, less cluttered with nested `if`s